src/librustdoc/lib.rs

```diff
@@ -777,9 +776,10 @@ fn main_options(options: config::Options) -> MainResult {
     let render_options = options.render_options.clone();
     let scrape_examples_options = options.scrape_examples_options.clone();
     let document_private = options.render_options.document_private;
+
     let config = core::create_config(options);
 
-    interface::create_compiler_and_run(config, |compiler| {
+    interface::run_compiler(config, |compiler| {
```

Comment:
One thing that strikes me is just how *many* different functions rustc_interface exposes with different names. Is anyone using just create_compiler_and_run` without thread locals? Can we make it private?

Reply:
You addressed this in a later commit.